pyk/src/pyk/kore/rpc.py

```diff
@@ -291,23 +291,24 @@ def close(self) -> None:
         self._transport.close()
 
     def request(self, method: str, **params: Any) -> dict[str, Any]:
-        old_id = self._req_id
+        # Generate unique ID regardless of different clients, repeated requests from the same client, etc.
+        uid = hash_str((id(self), self._req_id))
```

Comment:
I think I'm missing something. My understanding is that the request file names used to be a function of
* `id(self)` (through `self._bug_report_id`),
* `self._req_id`.

My questions are:
1. Which one of the two values causes duplication of file names, and how?
2. Variable `uid` seems to depend exactly on these two values. How does this ensure uniqueness?

Reply:
`self._bug_report_id` is not uniquely identifying the `JsonRpcClient`. We are currently just setting it as the proof ID, in @PetarMax's example, `rpc_contracts%model%TestContract.test_function():0`.

1. `self._req_id` causes the duplication of file names, because there are two separate `JsonRpcClient`s working on the same proof that each track their own `_req_id` starting from 1 and incrementing only when a request is submitted through that specific `JsonRpcClient`. (Their `_bug_report_id` is the same.)
2. It should be unique because the `id`s of different `JsonRpcClient` objects are different and within a single `JsonRpcClient`, `self._req_id` is still incremented with every new request, so the combination of the two should be unique across all requests from all `JsonRpcClient`s.

Reply:
Okay, so IIUC the root cause is, the caller tells the client what `_bug_report_id` to use, and that's not unique. Is this correct?

Reply:
Yes, I suppose it could be remedied instead by fixing the calling code so it is giving unique bug report IDs every time it's generating a new client.

Reply:
Or just redefine the role of `bug_report_id`: if given, the path within the bug report is

```
{bug_report_id}/{client_id}/request_{req_id}.json
```

and otherwise, it's

```
kore_rpc/{client_id}/request_{req_id}.json
```

or

```
{client_id}/request_{req_id}.json
```

Reply:
I've gone with this approach. The key thing is just to make sure the client ID is guaranteed included in the path. I see it wasn't previously when a `bug_report_id` was passed.